Index: po/Makefile
===================================================================
--- po/Makefile	(revision 5268)
+++ po/Makefile	(working copy)
@@ -28,12 +28,15 @@
 %.mo: %.po
 	$(BIN_MSGFMT) -o $@ $<
 
+# support installing only selected tranlsations
+ifdef PO
 install:
 	@$(BIN_MKDIR) -p $(addsuffix /LC_MESSAGES/, $(addprefix $(P_LOCALE)/, $(basename $(PO))))
 	@for i in $(basename $(PO)); do \
 		echo $(BIN_INSTALL) $$i.mo $(P_LOCALE)/$$i/LC_MESSAGES/$(GETTEXT_PACKAGE).mo; \
 		$(BIN_INSTALL) $$i.mo $(P_LOCALE)/$$i/LC_MESSAGES/$(GETTEXT_PACKAGE).mo; \
 	done
+endif
 
 
 CLEANFILES = $(MO)
