Index: src/subtitles_provider_libass.h
===================================================================
--- src/subtitles_provider_libass.h	(revision 5007)
+++ src/subtitles_provider_libass.h	(working copy)
@@ -42,7 +42,7 @@
 #include "stdint.h"
 #endif
 
-#include "../libass/ass.h"
+#include <ass/ass.h>
 }
 
 class FontConfigCacheThread;
Index: Makefile.inc.in
===================================================================
--- Makefile.inc.in	(revision 5007)
+++ Makefile.inc.in	(working copy)
@@ -96,8 +96,7 @@
 CFLAGS_FFMPEGSOURCE		= -I../libffms/include
 CFLAGS_HUNSPELL			= @HUNSPELL_CFLAGS@
 CFLAGS_ICONV			= @ICONV_CFLAGS@
-CFLAGS_LIBASS			= -I../libass
-#CFLAGS_LIBASS			= @LIBASS_CFLAGS@
+CFLAGS_LIBASS			= @LIBASS_CFLAGS@
 CFLAGS_LIBAVCODEC		= @LIBAVCODEC_CFLAGS@
 CFLAGS_LIBAVFORMAT		= @LIBAVFORMAT_CFLAGS@
 CFLAGS_LIBAVUTIL		= @LIBAVUTIL_CFLAGS@
@@ -119,8 +118,7 @@
 LDFLAGS_FFMPEGSOURCE	= ../libffms/libffmpegsource_aegisub.a
 LDFLAGS_HUNSPELL		= @HUNSPELL_LIBS@
 LDFLAGS_ICONV			= @ICONV_LDFLAGS@
-LDFLAGS_LIBASS			= ../libass/libass_aegisub.a
-#LDFLAGS_LIBASS			= @LIBASS_LIBS@
+LDFLAGS_LIBASS			= @LIBASS_LIBS@
 LDFLAGS_LIBAVCODEC		= @LIBAVCODEC_LIBS@
 LDFLAGS_LIBAVFORMAT		= @LIBAVFORMAT_LIBS@
 LDFLAGS_LIBAVUTIL		= @LIBAVUTIL_LIBS@
Index: Makefile.target
===================================================================
--- Makefile.target	(revision 5007)
+++ Makefile.target	(working copy)
@@ -121,7 +121,7 @@
 	$(MAKE) -C $@ $(MAKECMDGOALS)
 
 # Set relations to ensure dependencies are built before their targets during parallel builds.
-src: universalchardet libass libffms tools libaegisub
+src: universalchardet libffms tools libaegisub
 tests: libaegisub
 reporter: src
 
Index: Makefile
===================================================================
--- Makefile	(revision 5007)
+++ Makefile	(working copy)
@@ -1,9 +1,5 @@
 include Makefile.inc
 
-ifeq (yes, $(WITH_LIBASS))
-SUBDIRS += libass
-endif
-
 ifeq (yes, $(HAVE_PROVIDER_FFMPEGSOURCE))
 SUBDIRS += libffms
 endif
