Index: Makefile.target
===================================================================
--- Makefile.target	(revision 4897)
+++ Makefile.target	(revision 4952)
@@ -26,7 +26,7 @@
 ifdef LIB_SHARED
 LIB_SHARED_FULL = $(LIB_SHARED).$(LIB_VERSION)
 $(LIB_SHARED) : $(OBJ)
-	$(BIN_CXX) $(LIB_SHARED_LINK) $(OBJ) -o $(LIB_SHARED_FULL)
+	$(BIN_CXX) $(LIB_SHARED_LINK) $(LDFLAGS) $(OBJ) -o $(LIB_SHARED_FULL)
 	$(BIN_LN) -s $(LIB_SHARED_FULL) $(LIB_SHARED)
 
 CLEANFILES+= $(LIB_SHARED_FULL)
