diff --git a/configure b/configure
index 7aea31f..28b23cf 100755
--- a/configure
+++ b/configure
@@ -103,7 +103,7 @@ icl_ldflags() {
 
         arg=${arg/pthreadGC/pthreadVC}
         [ "$arg" = avifil32.lib ] && arg=vfw32.lib
-        [ "$arg" = gpac_static.lib ] && arg=libgpac_static.lib
+        [ "$arg" = gpac.lib ] && arg=libgpac.lib
 
         [ -n "$arg" ] && echo -n "$arg "
     done
@@ -823,7 +823,7 @@ fi
 
 if [ "$gpac" = "auto" ] ; then
     gpac="no"
-    cc_check "" -lz && GPAC_LIBS="-lgpac_static -lz" || GPAC_LIBS="-lgpac_static"
+    cc_check "" -lz && GPAC_LIBS="-lgpac -lz" || GPAC_LIBS="-lgpac"
     if [ "$SYS" = "WINDOWS" ] ; then
         GPAC_LIBS="$GPAC_LIBS -lwinmm"
     fi
